### ISSUE TITLE
Raw transforms of linked image should support custom link classes

### ISSUE THREAD

**Issue by greatislander** [WordPress/gutenberg#11995]
**Is your feature request related to a problem? Please describe.**

Now that image links can have a custom class assigned (see #7771), we should support existing linked images' link classes.

**Describe the solution you'd like**

Proposed solution identified by @talldan in https://github.com/WordPress/gutenberg/pull/7771#discussion_r231757707.

**Describe alternatives you've considered**

I have not explored alternatives yet.

**Comment by designsimply** [comment]
Copying the text from https://github.com/WordPress/gutenberg/pull/7771#discussion_r231757707 for reference:

> I'm not sure what the best practice is here, but it seems like you could define a regex that accepts any valid class, and that would make linkClass start working again for raw transforms.

@greatislander would you mind also including a short set of steps for testing for this issue?

**Comment by paaljoachim** [comment]
This issue can use a status update.
Thank you!

@tellthemachines @talldan

**Comment by talldan** [comment]
Still seems relevant. Classnames are still stripped from image link when pasted in the editor. To test, paste the following:
```html
<figure class="wp-block-image size-large"><a class="test" href="https://bbc.com"><img src="http://localhost:8888/wp-content/uploads/2021/02/Screenshot-2021-02-01-at-3.38.26-pm-2.png" alt="" class="wp-image-179"/></a></figure>
```

The class name 'test' will be lost on paste.

**Comment by paaljoachim** [comment]
This is an old issue. Is it still relevant?

**Comment by greatislander** [comment]
There's still data loss, so I would suggest that it's still relevant. However I don't actively follow the Gutenberg project anymore and don't use it so ¯\_(ツ)_/¯

**Comment by Mamaduka** [comment]
Here's the fix #67803.
